test/joystick.js

```diff
@@ -0,0 +1,301 @@
+var MockFirmata = require("./util/mock-firmata"),
+  five = require("../lib/johnny-five.js"),
+  sinon = require("sinon"),
+  Board = five.Board,
+  Joystick = five.Joystick;
+
+var io = new MockFirmata();
+var board = new Board({
+  io: io,
+  debug: false,
+  repl: false
+});
+
+
+exports["Joystick -- Analog"] = {
+
+  setUp: function(done) {
+
+    this.clock = sinon.useFakeTimers();
+    this.analogRead = sinon.spy(MockFirmata.prototype, "analogRead");
```

Comment:
Out of curiosity, why do you create a spy on the prototype instead of the instance?